app/src/features/UserProfileSettings/index.tsx

```diff
@@ -0,0 +1,420 @@
+import styled from 'styled-components';
+import { useSelector } from 'react-redux';
+import { selectIsUserLoggedIn, selectUser } from 'redux/auth/selectors';
+import Helmet from 'react-helmet';
+import { Col, Row } from 'react-bootstrap';
+import { HideAt, ShowAt } from 'react-with-breakpoints';
+import { Alert, Button, Divider, TextField } from '@mui/material';
+import {
+    ChangeUserNameRequest,
+    ChangeUserNameErrors,
+    ChangePasswordRequest,
+    ChangePasswordErrors,
+    RequestState,
+} from './types';
+import axiosAuthInstance from 'axios/instance';
+import { UPDATE_NAME, UPDATE_PASSWPRD } from 'constants/api';
+
+const getErrorsFromResponse = (errors: any) => {
+    if (!errors.response) {
+        return ['Something went wrong'];
+    }
+
+    const allErrors: any = [];
+
+    Object.values(errors.response.data.errors)
+        .map((e) => (Array.isArray(e) ? e : null))
+        .filter((e) => e !== null)
+        .forEach((e) => {
+            e?.forEach((ee) => allErrors.push(ee));
+        });
+
+    return allErrors;
+};
+
+const UserProfileSettings = () => {
+    const isLoggedIn = useSelector(selectIsUserLoggedIn);
+    const user = useSelector(selectUser);
+    const firstName = user?.firstName ?? '';
+    const lastName = user?.lastName ?? '';
+
+    const [changeUserNameRequest, setChangeUserNameRequest] =
+        useState<ChangeUserNameRequest>({
+            firstName,
+            lastName,
+        });
+
+    const [changePasswordRequest, setChangePasswordRequest] =
+        useState<ChangePasswordRequest>({
+            oldPassword: '',
+            newPassword: '',
+            repeatNewPassword: '',
+        });
+
+    useEffect(() => {
+        setChangeUserNameRequest({
+            firstName,
+            lastName,
+        });
+    }, [user]);
+
+    const nameValidationErrors = useMemo<ChangeUserNameErrors>(() => {
+        const errors: ChangeUserNameErrors = {
+            firstName: undefined,
+            lastName: undefined,
+        };
+        if (changeUserNameRequest.firstName.trim().length === 0) {
+            errors.firstName = 'First name is required';
+        }
+        if (changeUserNameRequest.lastName.trim().length === 0) {
+            errors.lastName = 'Last name is required';
+        }
+        return errors;
+    }, [changeUserNameRequest]);
+
+    const hasNameValidationErrors = useMemo(() => {
+        return (
+            nameValidationErrors.firstName !== undefined ||
+            nameValidationErrors.lastName !== undefined
+        );
+    }, [nameValidationErrors]);
+
+    const hasNameChanged =
+        changeUserNameRequest.firstName !== firstName ||
+        changeUserNameRequest.lastName !== lastName;
+
+    const passwordValidationErrors = useMemo<ChangePasswordErrors>(() => {
+        const errors: ChangePasswordErrors = {
+            oldPassword: undefined,
+            newPassword: undefined,
+            repeatNewPassword: undefined,
+        };
+        if (
```

Comment:
Cini mi se da ako ne popunimo nista uletece u ovaj if i vratice samo ovaj objekat sa svim vrednostima postavljenim na undefined

Reply:
Da li je to ono sto zelimo? xD

Reply:
undefined znaci nema greske, to sam uradio da ne bi greske postojale odmah cim udjes u formu

Reply:
s tim sto nisam ni to bas idealno resio jer gresaka ima odma cim pocnes da kucas stari password :(